File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnMoodgridResult.cs

namespace GracenoteSDK {

/**
** <b>Experimental</b>: GnMoodgridResult
*/
public class GnMoodgridResult : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnMoodgridResult(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnMoodgridResult obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnMoodgridResult() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnMoodgridResult(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

/**
*  Returns the count of the GnMoodgridIdentifier instances in this result.
* @return count
*/
  public uint Count() {
    uint ret = gnsdk_csharp_marshalPINVOKE.GnMoodgridResult_Count(swigCPtr);
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
    return ret;
  }

  public GnMoodgridResultEnumerable Identifiers {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnMoodgridResult_Identifiers_get(swigCPtr);
      GnMoodgridResultEnumerable ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnMoodgridResultEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnMoodgridResult.java

package com.gracenote.gnsdk;

/** 
** <b>Experimental</b>: {@link GnMoodgridResult} 
*/ 
 
public class GnMoodgridResult {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnMoodgridResult(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnMoodgridResult obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnMoodgridResult(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

/** 
*  Returns the count of the {@link GnMoodgridIdentifier} instances in this result. 
* @return count 
*/ 
 
  public long count() throws com.gracenote.gnsdk.GnException {
    return gnsdk_javaJNI.GnMoodgridResult_count(swigCPtr, this);
  }

/** 
* Retrieves an iterator to access the {@link GnMoodgridIdentifier} instances in this result. 
* @return identifiers 
*/ 
 
  public GnMoodgridResultIterable identifiers() {
    return new GnMoodgridResultIterable(gnsdk_javaJNI.GnMoodgridResult_identifiers(swigCPtr, this), true);
  }

}
